src/charm.py

```diff
@@ -127,6 +128,19 @@ def _get_external_hostname(self) -> str:
             self.config["external_hostname"] if self.config["external_hostname"] else self.app.name
         )
 
+    def _is_setup_completed(self) -> bool:
+        """Check if the _set_up_discourse process has finished.
+
+        Returns:
+            True if the _set_up_discourse process has finished.
+        """
+        return SETUP_COMPLETED_FLAG_FILE.is_file()
+
+    def _set_setup_completed(self) -> None:
+        """Mark the _set_up_discourse process as completed."""
+        SETUP_COMPLETED_FLAG_FILE.parent.mkdir(exist_ok=True)
+        pathlib.Path("/run/discourse-k8s-operator/setup_completed").touch(exist_ok=True)
```

Comment:
Should this use the constant from the top of the file?

Reply:
Yes, updated, thanks!